src/tools/UBGraphicsRuler.cpp

```diff
@@ -466,6 +466,14 @@ void UBGraphicsRuler::mouseReleaseEvent(QGraphicsSceneMouseEvent *event)
     }
     else
     {
+        // snap to grid
```

Comment:
Shouldn't it be handled in the mouseMoveEvent instead ?

Reply:
That is not so easy here, because moving the tools is actually not handled by OpenBoard code, but left to Qt:

https://github.com/OpenBoard-org/OpenBoard/blob/1665e2e570e5e76c75d9e8a4b7e9a3f0c5071736/src/tools/UBGraphicsRuler.cpp#L420-L423

In such a case we would either have to modify much more code and implement the moving by ourselves, or modify the event, what I think is never a good idea. So I did the snapping here (and also for other tools) when releasing the mouse button.

T know that this leads to different behavior for different cases, but personally found that this does not hurt my experience.

Reply:
As I didn't implemented it and didn't know what to expect, I found it very counter-intuitive, and tried three of four times wondering what I was doing wrong with the tools, I only realized it was actually snapping after looking at the code. 

This snap feature is quite important imo, and I think it is very important that its behavior is harmonized, for moving as for rotating. I understand that it implies more code rework, but I think it is necessary, to guarantee a good user experience.

Reply:
> This snap feature is quite important imo, and I think it is very important that its behavior is harmonized, for moving as for rotating. I understand that it implies more code rework, but I think it is necessary, to guarantee a good user experience.

Ok, I understand. I think this affects all the C++ tools.

Reply:
It was much easier than expected and in fact, I can just move the code to `mouseMoveEvent`. For the compass, I also improved snapping of the pencil when resizing. It now also rotates the compass, so that you see exactly through which grid point the circle will go.

I have added the changes as a set of fixup commits and will merge them with the original commits before merging this PR. to keep the clean commit structure. Please tell me when you and your team are satisfied with the behavior.